### ISSUE TITLE
[Bug] Migration fails on a new system

### ISSUE THREAD

**Issue by chevdor** [AppFlowy-IO/AppFlowy#6583]
### Bug Description

Deploying on a fresh database.
```
Error: Failed to initialize application state: Failed to run migrations: while executing migration 20231130140000: error returned from database: relation "auth.users" does not exist
```

### How to Reproduce

- new postgres db
- using `postgres://postgres:somepass@postgresql:5432/appflowy`

### Expected Behavior

The migration works.

### Operating System

K8S

### AppFlowy Version(s)

appflowyinc/appflowy_cloud:latest

### Screenshots

<img width="1078" alt="image" src="https://github.com/user-attachments/assets/7e848024-c98b-4a38-bd2a-6cb235a4584b">

**Comment by khorshuheng** [comment]
Please see: https://github.com/AppFlowy-IO/AppFlowy-Cloud/issues/861

**Comment by chevdor** [comment]
Here is the mirgation: https://github.com/AppFlowy-IO/AppFlowy-Cloud/blob/main/migrations/20231130140000_user_delete_trigger.sql

**Comment by chevdor** [comment]
@khorshuheng thx for such a speedy pointer, that seems to be indeed spot on...

**Comment by khorshuheng** [comment]
There are actually some additional steps needed if you want to set this up on k8s. There's a work in progress here (which is not fully working yet) - https://github.com/khorshuheng/appflowy-self-host-resources/tree/main/charts/appflowy

**Comment by khorshuheng** [comment]
I am still investigating if we can simplify the setup via: https://github.com/AppFlowy-IO/AppFlowy-Cloud/pull/891

**Comment by chevdor** [comment]
I did find the script mentioned in the other issue but that was indeed leading me to the next issue you described.
Thanks for laying all of that down ! I did not expect some of the assumption (ie that the database must be named `postgres`) but that's not a big deal for now.

For now, I am using a manual deploy so I consider I still fall under the docker umbrella :)
I will check your chart.

**Comment by khorshuheng** [comment]
In my chart, you will see that I am using "appflowy" user and "appflowy" database. But it's still not fully working, because AppFlowy cloud service will fail to create user's data after login, due to row level security (which can be overcome with the draft PR above)

**Comment by chevdor** [comment]
Yes, that sounds already better than assuming the user can use the postgres DB...
I don't mind that for now though as long as I can get the db to a usage state. You write up is great, I am following your steps....

**Comment by chevdor** [comment]
From what I have seen so far, there are a few issues with appflowy_cloud but that's manageable.
There are many more issues due to `gotrue` regarding some migrations that no longer seem to be in the repo: 2021..2023 but I am not through yet.

The issues are mostly (if not only) around `auth.users` and [before](https://github.com/AppFlowy-IO/AppFlowy-Cloud/tree/main/migrations/before) does not seem to contain everything that is needed for `gotrue` to successfully apply all migrations. 

Why were the old migrations removed ?

**Comment by khorshuheng** [comment]
Basically, gotrue service will create the necessary migrations on start up. There are certain operations that cannot be handled automatically by gotrue, such as creating the auth schema (https://github.com/AppFlowy-IO/AppFlowy-Cloud/blob/ba8a670101d0a20fcad6dd2825f4d9561551fde5/migrations/before/20230312043000_supabase_auth.sql#L33), granting access to the auth schema to the supabase_auth_admin user, and creating some additional roles. That's why they are in the `before` folder.

Without seeing the detail error message, i can only speculate. Most of the users faced the issue due to one or more of the following reasons:
1. The `auth` schema is not created (CREATE SCHEMA IF NOT EXISTS auth AUTHORIZATION supabase_auth_admin)
2. `auth` schema is created but the default search path for the gotrue database user isn't set to `auth` (ALTER USER supabase_auth_admin SET search_path = 'auth';)
3. Using a different username and password for gotrue database user (if you use the default migration script, it should be supabase_auth_admin, and the password is `root`)

**Comment by khorshuheng** [comment]
As for your questions on old migrations - as explained, the migration is handled by the gotrue container. If you exec into the gotrue container, you will see all the migration files in that container, similar to https://github.com/supabase/auth/tree/master/migrations

**Comment by chevdor** [comment]
If only I could....

<img width="633" alt="image" src="https://github.com/user-attachments/assets/a4b99be6-0c8f-44b2-97bf-b903dee2a142">

Seeing [the image](https://github.com/AppFlowy-IO/AppFlowy-Cloud/blob/main/docker/gotrue.Dockerfile), it makes sense that `exec` does not work. But I see where those migrations are.
In my case, one of the issue is likely that I ran the `appflowy_cloud` migrations before the `gotrue` ones.

**Comment by chevdor** [comment]
If anyone else runs into that, here are the [`gotrue` migrations](https://github.com/supabase/auth/tree/master/migrations)
